browser/brave_cosmetic_resources_tab_helper.cc

```diff
@@ -0,0 +1,457 @@
+    }
+    if (window.content_cosmetic.generichide === undefined) {
+        window.content_cosmetic.generichide = %s;
+    })";
+
+const char kSelectorsInjectScript[] =
+    R"((function() {
+       let nextIndex =
+          window.content_cosmetic.cosmeticStyleSheet.rules.length;
+       const selectors = %s;
+       selectors.forEach(selector => {
+         if (!window.content_cosmetic.allSelectorsToRules.has(selector)) {
+           let rule = selector + '{display:none !important;}';
+           window.content_cosmetic.cosmeticStyleSheet.insertRule(
+             `${rule}`, nextIndex);
+           window.content_cosmetic.allSelectorsToRules.set(
+             selector, nextIndex);
+           nextIndex++;
+           window.content_cosmetic.firstRunQueue.add(selector);
+         }
+       });
+       if (!document.adoptedStyleSheets.includes(
+           window.content_cosmetic.cosmeticStyleSheet)) {
+         document.adoptedStyleSheets =
+           [window.content_cosmetic.cosmeticStyleSheet];
+       };
+    })();)";
+
+const char kStyleSelectorsInjectScript[] =
+    R"((function() {
+      let nextIndex =
+          window.content_cosmetic.cosmeticStyleSheet.rules.length;
+      const selectors = %s;
+      for (let selector in selectors) {
+        if (!window.content_cosmetic.allSelectorsToRules.has(selector)) {
+          let rule = selector + '{';
+          selectors[selector].forEach(prop => {
+            if (!rule.endsWith('{')) {
+              rule += ';';
+            }
+            rule += prop;
+          });
+          rule += '}';
+          window.content_cosmetic.cosmeticStyleSheet.insertRule(
+            `${rule}`, nextIndex);
+          window.content_cosmetic.allSelectorsToRules.set(
+            selector, nextIndex);
+          nextIndex++;
+        };
+      };
+      if (!document.adoptedStyleSheets.includes(
+            window.content_cosmetic.cosmeticStyleSheet)){
+         document.adoptedStyleSheets =
+           [window.content_cosmetic.cosmeticStyleSheet];
+      };
+    })();)";
+
+bool ShouldDoCosmeticFiltering(content::WebContents* contents,
+                               const GURL& url) {
+  Profile* profile = Profile::FromBrowserContext(contents->GetBrowserContext());
+  auto* map = HostContentSettingsMapFactory::GetForProfile(profile);
+
+  return ::brave_shields::ShouldDoCosmeticFiltering(map, url);
+}
+
+std::string LoadDataResource(const int id) {
+  auto& resource_bundle = ui::ResourceBundle::GetSharedInstance();
+  if (resource_bundle.IsGzipped(id)) {
+    return resource_bundle.LoadDataResourceString(id);
+  }
+
+  return resource_bundle.GetRawDataResource(id).as_string();
+}
+
+std::unique_ptr<base::ListValue> GetUrlCosmeticResourcesOnTaskRunner(
+    const std::string& url) {
+  auto result_list = std::make_unique<base::ListValue>();
+
+  base::Optional<base::Value> resources =
+      g_brave_browser_process->ad_block_service()->UrlCosmeticResources(url);
```

Comment:
this is an unsafe access to ad_block_service, ad_block_regional_service_manager and ad_block_custom_filters_service outside the UI thread